test/puppetlabs/puppetdb/testutils/services.clj

```diff
@@ -110,11 +110,10 @@
 (defmacro with-puppetdb-instance
   "Convenience macro to launch a puppetdb instance"
   [& body]
-  `(with-redefs [sutils/db-metadata (delay {:database "HSQL Database Engine"
-                                            :version [2 2]})]
+  `(with-redefs [sutils/db-metadata (delay (sutils/db-metadata-fn))]
```

Comment:
addressed in the channel -- it's a delay because it redefs a delay and is derefed as a delay. It could be an atom, but we'd be going out of our way.